ui/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/MockFetchingProcessRepository.scala

```diff
@@ -1,96 +1,118 @@
+    getUserProcesses[PS].map(_.find(p => p.idWithName.id == processId && p.processVersionId == versionId))
 
-    override def fetchProcessName(processId: ProcessId)(implicit ec: ExecutionContext): Future[Option[ProcessName]] =
-      Future(processes.find(p => p.processId == processId).map(_.idWithName.name))
+  override def fetchProcessId(processName: ProcessName)(implicit ec: ExecutionContext): Future[Option[ProcessId]] =
+    Future(processes.find(p => p.idWithName.name == processName).map(_.processId))
 
-    override def fetchProcessingType(processId: ProcessId)(implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[String] =
-      getUserProcesses.map(_.find(p => p.processId == processId).map(_.processingType).get)
+  override def fetchProcessName(processId: ProcessId)(implicit ec: ExecutionContext): Future[Option[ProcessName]] =
+    Future(processes.find(p => p.processId == processId).map(_.idWithName.name))
 
-    override def fetchProcesses[PS: processdetails.ProcessShapeFetchStrategy](isSubprocess: Option[Boolean], isArchived: Option[Boolean], isDeployed: Option[Boolean], categories: Option[Seq[String]], processingTypes: Option[Seq[String]])(implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[List[processdetails.BaseProcessDetails[PS]]] =
-      getUserProcesses[PS].map(_.filter(
-        p => check(isSubprocess, p.isSubprocess) && check(isArchived, p.isArchived) && check(isDeployed, p.isDeployed) && checkSeq(categories, p.processCategory) && checkSeq(processingTypes, p.processingType)
-      ))
+  override def fetchProcessingType(processId: ProcessId)(implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[String] =
+    getUserProcesses[Unit].map(_.find(p => p.processId == processId).map(_.processingType).get)
 
-    //TODO: Implement
-    override def fetchLatestProcessVersion[PS: processdetails.ProcessShapeFetchStrategy](processId: ProcessId)(implicit loggedUser: LoggedUser): Future[Option[ProcessVersionEntityData]] =
-      Future(None)
+  override def fetchProcesses[PS: processdetails.ProcessShapeFetchStrategy](isSubprocess: Option[Boolean], isArchived: Option[Boolean], isDeployed: Option[Boolean], categories: Option[Seq[String]], processingTypes: Option[Seq[String]])(implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[List[processdetails.BaseProcessDetails[PS]]] =
+    getUserProcesses[PS].map(_.filter(
+      p => check(isSubprocess, p.isSubprocess) && check(isArchived, p.isArchived) && check(isDeployed, p.isDeployed) && checkSeq(categories, p.processCategory) && checkSeq(processingTypes, p.processingType)
+    ))
 
-    //TODO: Implement
-    override def fetchProcessActions(processId: ProcessId)(implicit ec: ExecutionContext): Future[List[processdetails.ProcessAction]] =
-      Future(List.empty)
+  //TODO: Implement
```

Comment:
can we have ??? instead of Future(...) as implementation? Or throw an exception? So it's clear it's not implemented yet?

Reply:
Done